doc/architecture/adr-004.md

```diff
@@ -0,0 +1,82 @@
+# Transition to Cosmos SDK's NFT module
+
+## Changelog
+
+- 04-11-2021: Initial Draft
+
+## Context
+
+Currently, Crypto.com Chain offers support for non-fungible tokens (NFTs) using a custom NFT module in `chain-main`. The
+current implementation of NFT module is based on [Irismod's NFT module](https://github.com/irisnet/irismod/tree/master/modules/nft).
+Given that different chain's are using their custom implementation of NFT module, it is very difficult to achieve
+flawless interportability of NFTs across multiple chains using IBC. At the same time, there are more usecases that needs
+to be supported using NFTs which may require IBC support. So, there is a need of a standardized NFT implementation
+which has support for reusing NFTs between modules and chains.
+
+## Decision
+
+There are multiple ways to address above mentioned issues.
+
+1. Use Cosmos SDK's NFT module
+
+   To solve interportablity and standardisation issues, we can adopt [Cosmos SDK's NFT module](https://github.com/cosmos/cosmos-sdk/blob/master/docs/architecture/adr-043-nft-module.md)
+   which is expected to become standard for implementing NFTs on Cosmos SDK chains and also expected to get support for
+   transferring NFTs via IBC. 
+
+   > More details on interportability of Cosmos SDK's NFT module [here](https://github.com/cosmos/cosmos-sdk/blob/master/docs/architecture/adr-043-nft-module.md#interoperability)
+
+   **Advantages**
+
+   - Because this module is a part of Cosmos SDK, it'll get all the future improvement for free.
+   - Much easier upgrades to newer versions of NFT module as the upgrades will be tested extensively by the community.
+
+   **Drawbacks**
+
+   - It'll become a bit difficult to add custom logic (especially when transferring NFTs) based on our requirements.
+
+2. Modify current NFT implementation in `chain-main`
+   
+   Cosmos SDK's NFT module [ADR](https://github.com/cosmos/cosmos-sdk/blob/master/docs/architecture/adr-043-nft-module.md)
+   outlines the API contract implemented by NFT module.
+
+   **Advantages**
+
+   - It'll be very easy to add custom logic base on our requirements (for example, https://github.com/irisnet/irismod/issues/183).
+
+   **Drawbacks**
+
+   - Because this module will be developed in isolation, we'll have to work extra hard to make sure that it confirms to
+     Cosmos SDK's NFT standards.
+   - Custom logic added to the module may make it impossible to stay in sync with Cosmos SDK's standards.
```

Comment:
given this is not a decision to take, it can be removed from here

Reply:
I wanted to keep it here so that in future, we know all the things that we considered.